components/movementsensor/gpsutils/vrs.go

```diff
@@ -3,32 +3,50 @@ package gpsutils
 
 import (
 	"bufio"
+	"context"
 	"encoding/base64"
+	"errors"
 	"fmt"
+	"io"
 	"net"
 	"net/url"
+	"time"
 
 	"go.viam.com/rdk/logging"
+	"go.viam.com/rdk/utils"
 )
 
+const (
+	vrsGGARateSec = 20 // rate to send GGA messages to the VRS in seconds
+)
+
+// VRS contains the VRS.
+type VRS struct {
+	ntripInfo    *NtripInfo
+	readerWriter *bufio.ReadWriter
+	conn         net.Conn
+	workers      utils.StoppableWorkers
+	logger       logging.Logger
+}
+
 // ConnectToVirtualBase is responsible for establishing a connection to
 // a virtual base station using the NTRIP protocol with enhanced error handling and retries.
-func ConnectToVirtualBase(ntripInfo *NtripInfo, logger logging.Logger) (*bufio.ReadWriter, net.Conn, error) {
+func ConnectToVirtualBase(ctx context.Context, ntripInfo *NtripInfo, logger logging.Logger) (*VRS, error) {
```

Comment:
It looks to me like `ctx` is unused. Why pass it in? Any chance it can be removed again?

Reply:
it was initially used for the cancel context, so the VRS would cancel its context with the gpsrtk/parent

Reply:
ended up keeping `ctx` for that for loop